FirebaseAppDistribution/Sources/FIRAppDistributionUIService.m

```diff
@@ -68,10 +68,20 @@ - (void)appDistributionRegistrationFlow:(NSURL *)URL
         callbackURLScheme:callbackURL
         completionHandler:^(NSURL *_Nullable callbackURL, NSError *_Nullable error) {
           [self resetUIState];
-          FIRFADInfoLog(@"Sign in complete using ASWebAuthenticationSession");
-          // TODO (b/161538029): Map these errors to AppDistribution error codes
-
-          completion(error);
+          [self logRegistrationCompletion:error authType:[ASWebAuthenticationSession description]];
+          if (error) {
```

Comment:
Can we extract this as a method so that we don't have to repeat this code? You could even have the error handler return error or nil and just ` return completion([self getAppDistributionError:appDistributionError])`. Maybe something like:
```
+ (NSError *_Nullable)mapErrorAppDistributionError:(NSError *_Nullable)error{
    if(!error) {
       return nil;
    }
    
    if([error code] == ASWebAuthenticationSessionErrorCodeCanceledLogin || [error code]  == SFAuthenticationErrorCanceledLogin) {
        return [self getAppDistribution: FIRAppDistributionErrorAuthenticationCancelled];
    }
    return [self getAppDistribution: FIRAppDistributionErrorAuthenticationFailure];
}
```

Reply:
I thought about doing that. But that would mean repeating the ios version availability code again for `ASWebAuthenticationSessionErrorCodeCanceledLogin`  and `SFAuthenticationErrorCanceledLogin `. Neither options felt that great so I decided to go with repeating the logging logic.

Reply:
Why would we need to duplicate that piece of the logic? I think I'm misunderstanding something here.